daemon/networkdriver/ipallocator/allocator.go

```diff
@@ -41,19 +41,24 @@ var (
 	ErrBadSubnet                = errors.New("network does not contain specified subnet")
 )
 
-var (
-	lock         = sync.Mutex{}
-	allocatedIPs = networkSet{}
-)
+type IPAllocator struct {
+	allocatedIPs networkSet
+	sync.Mutex
```

Comment:
can we not expose `sync.Mutex`, because `Lock/Unlock` methods for ipallocator look weird.